### PR TITLE
[PM-386] Add deploy/code-review pipeline for buyerbank

### PR DESCRIPTION
**Pull request by MarcoAGreco**
### List of changes
- Add code-review pipeline for pagopa-buyerbank Azure functions
- Add  deploy pipeline for pagopa-buyerbank Azure functions

### Motivation and context

- New product
### Type of changes

- [X] Add new pipeline
- [ ] Update pipeline configuration
- [ ] Remove pipeline

### :warning: If it's new pipeline with code review have you added pagopa-github-bot -> Role: admin?

- [x] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: azure-devops/projects/pagopa-projects/pagopa-functions-buyerbank.tf [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  # code_review secrets
+  pagopa-functions-buyerbank-variables_secret_code_review = {
+
+  }
+  # deploy vars
+  pagopa-functions-buyerbankvariables_deploy = {
+    git_mail                = module.secrets.values["io-azure-devops-github-EMAIL"].value
+    git_username            = module.secrets.values["io-azure-devops-github-USERNAME"].value
+    github_connection       = azuredevops_serviceendpoint_github.io-azure-devops-github-rw.service_endpoint_name
+    uat_azure_subscription  = azuredevops_serviceendpoint_azurerm.UAT-PAGOPA.service_endpoint_name
```
Comment (infantesimone) [inline azure-devops/projects/pagopa-projects/pagopa-functions-buyerbank.tf:40]: @MarcoAGreco add also dev subscription:
-  dev_azure_subscription  = azuredevops_serviceendpoint_azurerm.DEV-PAGOPA.service_endpoint_name

**Review by infantesimone — APPROVED** [review]

**Review by uolter — DISMISSED** [review]

**Comment by uolter** [comment]
@MarcoAGreco the apply fails with the following error:

```
module.pagopa-functions-buyerbank_code_review[0].azuredevops_build_definition.pipeline: Creating...

Error: error creating resource Build Definition: Unable to configure a service on the selected GitHub repository. This is likely caused by not having the necessary permission to manage hooks for the selected repository.

  on .terraform/modules/pagopa-functions-buyerbank_code_review/azuredevops_build_definition_code_review/main.tf line 18, in resource "azuredevops_build_definition" "pipeline":
  18: resource "azuredevops_build_definition" "pipeline" {
```

some permission is missed at the github repo. I think you can compare that function app with another repo like function checkout.

File: azure-devops/projects/pagopa-projects/pagopa-functions-buyerbank.tf [hunk 2]
```diff
@@ -0,0 +1,97 @@
+variable "pagopa-functions-buyerbank" {
+  default = {
+    repository = {
+      organization    = "pagopa"
+      name            = "pagopa-functions-buyerbank"
+      branch_name     = "main"
```
Comment (infantesimone) [inline azure-devops/projects/pagopa-projects/pagopa-functions-buyerbank.tf:6]: @MarcoAGreco master (and not main)

**Review by uolter — APPROVED** [review]

**Comment by pasqualedevita** [comment]
> @MarcoAGreco the apply fails with the following error:
> 
> ```
> module.pagopa-functions-buyerbank_code_review[0].azuredevops_build_definition.pipeline: Creating...
> 
> Error: error creating resource Build Definition: Unable to configure a service on the selected GitHub repository. This is likely caused by not having the necessary permission to manage hooks for the selected repository.
> 
>   on .terraform/modules/pagopa-functions-buyerbank_code_review/azuredevops_build_definition_code_review/main.tf line 18, in resource "azuredevops_build_definition" "pipeline":
>   18: resource "azuredevops_build_definition" "pipeline" {
> ```
> 
> some permission is missed at the github repo. I think you can compare that function app with another repo like function checkout.

@MarcoAGreco @infantesimone please check also pagopa-github-bot -> Role: admin

**Pull request merged**